tiledb/sm/config/config.h

```diff
@@ -595,6 +599,22 @@ class Config {
   template <class T>
   Status get(const std::string& param, T* value, bool* found) const;
 
+  /**
+   * Retrieves a string value of the given parameter.
+   * Throws Status_ConfigError if config value could not be found
+   */
+  std::string get(
```

Comment:
It would be better to leave the non-template one out entirely. Its analogue has been documented as deprecated.

You can specialize the template for `std::string` and use the body that had been here for the specialization. That gives the basic utility while preserving the named-type syntax.

Reply:
Still a WIP, but I pushed where I'm at currently. I removed this function but eventually ended up adding a similar one back for `template<bool must_find> string get_private()` as I was getting errors around partial template specializations against `template<T, bool must_find> optional<T> get_private()`. I'll keep looking at this tomorrow but LMK if you have any other thoughts or suggestions.